src/svg/gedaelementlexer.cpp

```diff
@@ -25,27 +25,32 @@ along with Fritzing.  If not, see <http://www.gnu.org/licenses/>.
 
 static QRegExp findWhitespace("[\\s]+");
 
-GedaElementLexer::GedaElementLexer(const QString &source)
+GedaElementLexer::GedaElementLexer(const QString &source) :
+    m_nonWhitespaceMatcher("[^\\s]"),
+	m_commentMatcher("(^\\s*\\#)"),
+	m_elementMatcher("Element\\s*([\\(\\[])"),
+	m_stringMatcher("\"([^\"\\\\]*(\\\\.[^\"\\\\]*)*)\""),
+	m_integerMatcher("[-+]?\\d+"),
+    m_hexMatcher("0[xX][0-9a-fA-F]+"),
+    m_source(),
+    m_chars(nullptr),
+    m_size(0),
+    m_pos(0),
+    m_current(0),
+    m_currentCommand(),
+    m_currentNumber(0l),
+    m_currentHexString(0l),
+    m_currentString(),
+    m_comments()
 {
-	m_nonWhitespaceMatcher.setPattern("[^\\s]");
-	m_commentMatcher.setPattern("(^\\s*\\#)");
-	m_elementMatcher.setPattern("Element\\s*([\\(\\[])");
-	//m_stringMatcher.setPattern("\"([^\"]*)\"");
-	m_stringMatcher.setPattern("\"([^\"\\\\]*(\\\\.[^\"\\\\]*)*)\"");
-	m_integerMatcher.setPattern("[-+]?\\d+");
-	m_hexMatcher.setPattern("0[xX][0-9a-fA-F]+");
+    /// @todo figure out how to do inplace initialization of a cleaned source string.
```

Comment:
The clean() method is quite expensive, compared to the QString copy/move operation, maybe this todo is a bit overkill?

Reply:
With guaranteed copy elision in c++11/14, the value returned is actually the memory within the instance. You're right that clean is very expensive. That is exactly why I added the todo, I figured that it may be beneficial but I wasn't totally convinced. Just didn't want to forget about it.